catalog/app/utils/json-schema.js

```diff
@@ -0,0 +1,84 @@
+import * as R from 'ramda'
+import isArray from 'lodash/isArray'
+import isNumber from 'lodash/isNumber'
+import isObject from 'lodash/isObject'
+import isString from 'lodash/isString'
+import isUndefined from 'lodash/isUndefined'
+
+function isSchemaArray(optSchema) {
+  return R.prop('type', optSchema) === 'array'
+}
+
+function isSchemaObject(optSchema) {
+  return R.prop('type', optSchema) === 'object'
+}
+
+function isSchemaString(optSchema) {
+  return R.prop('type', optSchema) === 'string'
+}
+
+function isSchemaNumber(optSchema) {
+  return R.prop('type', optSchema) === 'number'
+}
+
+function isSchemaBoolean(optSchema) {
+  return R.prop('type', optSchema) === 'boolean'
+}
+
+function isSchemaNull(optSchema) {
+  return R.prop('type', optSchema) === 'null'
+}
+
+export function isSchemaEnum(optSchema) {
+  return Boolean(R.prop('enum', optSchema))
```

Comment:
`R.propSatisfies('enum', Boolean)`

Reply:
I've just tried this. It throws error when `optSchema` is undefined. Same as `R.has` and `R.propEq`:

`R.prop('enum', undefined)` --> `undefined`
`R.has('enum', undefined)` --> throws error
`R.propEq('type', 'array', undefined)` --> throws error
`R.propSatisfies('enum', Boolean, undefined)` --> throws error